react/week4/github_users_search/src/index.js

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import UserList from "./UserList";
+//import App from './App';
+import searchContext from "./context";
+import * as serviceWorker from './serviceWorker';
+
+const root = document.getElementById('root');
+const ContextProvider = searchContext.Provider;
+
+function fetchGithubUsers(query) {
+  if (query !== '') {
+    const URL = `https://api.github.com/search/users?q=${query}`
+    return fetch(URL).then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        return response.json();
+      } else {
+        throw Error(response.statusText);
+      }
+    })
+  }
+}
+
+function fetchGithubUsersRepos(login) {
+  const URL = `https://api.github.com/users/${login}/repos`
+  return fetch(URL).then((response) => {
+    if (response.status >= 200 && response.status <= 299) {
+      return response.json();
+    } else {
+      throw Error(response.statusText);
+    }
+  })
+}
+
+function Header() {
+  return (
+    <div>
+      <h1>Github user searcher</h1>
+    </div>
+  )
+}
+
+class Searchbar extends React.Component {
+  state = {
+    searchTerm: '',
+    isLoading: false,
+    usersArray: [],
+    reposArray: [],
+    errorMessage: '',
+    isClickedUser: false,
+    isClickedRepos: false,
+    selectedUser: ''
+  }
+
+  handleSearchChange = (event) => {
+    const searchedValue = event.target.value
+    this.setState({
+      searchTerm: searchedValue
+    })
+    if (searchedValue !== '') {
+      fetchGithubUsers(searchedValue)
+        .then(data => {
+          console.log(data);
+          const users = data.items.map(user => user)
+          console.log(users);
+          this.setState({
+            usersArray: users,
+            isLoading: true
+          })
+        }).catch(error => {
+          console.log(error.message);
+          this.setState({
+            errorMessage: error.message,
+            usersArray: []
+          })
+        }).finally(() => this.setState({ isLoading: false }))
+    }
+  }
+
+  handleClickSearch = (event, loginName) => {
+    event.preventDefault();
+    console.log(loginName);
+    fetchGithubUsersRepos(loginName)
+      .then(data => {
+        const repos = data.map(repo => repo)
```

Comment:
Good use of `const`.

Also, the map `data.map(repo => repo)` is semantically equivalent to `data`. `.map(repo => repo)` is what is known as an [identity function](https://en.wikipedia.org/wiki/Identity_function).